### PR TITLE
UML-1507 : Changes that dashboard doesn't break when donor date of birth is  missing

### PR DESCRIPTION
**Pull request by SeemaMenon**
# Purpose

https://opgtransform.atlassian.net/browse/UML-1507

Fixes UML-1507

## Approach

_Explain how your code addresses the purpose of the change_

## Learning

_Any tips and tricks, blog posts or tools which helped you. Plus anything notable you've discovered about the Use a Lasting Power of Attorney service_

## Checklist

* [X] I have performed a self-review of my own code
* [ ] I have added relevant logging with appropriate levels to my code
* [ ] I have updated documentation (Confluence/GitHub wiki/tech debt doc) where relevant
* [X] I have added tests to prove my work
* [ ] I have added welsh translation tags and updated translation files
* [ ] I have run an accessibility tool on any pages I have made changes to and fixed any issues found
* [ ] The product team have tested these changes

### CODE REVIEW

File: service-api/app/src/App/src/Service/Lpa/OlderLpaService.php [hunk 1]
```diff
@@ -67,6 +67,18 @@ public function cleanseUserData(array $data): ?array
 
     public function checkDataMatch(array $actor, array $userDataToMatch): ?array
     {
+        //check if actor has a valid dob
```
Comment (GemTay) [inline service-api/app/src/App/src/Service/Lpa/OlderLpaService.php:70]: if the donors DOB on the LPA is blank, it wont match what they have entered anyway, so this code isnt needed

File: service-api/app/src/App/src/Service/Lpa/OlderLpaService.php [hunk 2]
```diff
@@ -212,6 +224,12 @@ public function checkLPAMatchAndGetActorDetails(array $dataToMatch): array
         }
 
         if (!($this->validateLpaRequirements)($lpaMatchResponse->getData())) {
+            $this->logger->info(
```
Comment (GemTay) [inline service-api/app/src/App/src/Service/Lpa/OlderLpaService.php:227]: we already log this in the validateLpaRequirements class

**Review by GemTay — CHANGES_REQUESTED** [review]
When users create a code and give it to an organisation, the LPA summary on the viewer side will either break or the donors DOB will be blank. I'm not sure how organisations will feel about this and whether it means the LPA can still be used. I think we need to go back to Sarah and see how we'd like to deal with this

File: service-api/app/src/App/src/Service/Lpa/OlderLpaService.php (hunk 2 above)
Comment (SeemaMenon) [reply service-api/app/src/App/src/Service/Lpa/OlderLpaService.php:227]: you are right .... removed.

File: service-api/app/src/App/src/Service/Lpa/OlderLpaService.php (hunk 1 above)
Comment (SeemaMenon) [reply service-api/app/src/App/src/Service/Lpa/OlderLpaService.php]: Yeah, get you there.

I am so tempted to not do a data match check if the relevant details on the LPA are not available.

**Comment by SeemaMenon** [comment]
> When users create a code and give it to an organisation, the LPA summary on the viewer side will either break or the donors DOB will be blank. I'm not sure how organisations will feel about this and whether it means the LPA can still be used. I think we need to go back to Sarah and see how we'd like to deal with this

The viewer side  [and the actor side] would not break, but would show a blank entry for the dob.
This has been discussed [with Jude in Sarah's absence] and I think for now, business is OK with a blank entry  for DB in view summary[Than have a breaking dashboard on the actor side].Is mentioned in the ticket I think. 

What Jude mentioned to me was the new ticket 1588 would be the new feature to make users aware and be proactive.
I also think it's a data cleansing  activity which is in discussion.

**Review by GemTay — APPROVED** [review]

**Pull request merged**